core/commands/diff.py

```diff
@@ -468,54 +484,198 @@ class GsDiffOpenFileAtHunkCommand(TextCommand, GitCommand):
-            if use_prepix is None:
-                filename = diff_start_line.split(" ")[-1]
-            else:
-                filename = use_prepix.groups()[0]
-
-            # Example: "@@ -9,6 +9,7 @@" --> 9
-            lineno = int(re.search(r"^@@ \-\d+(,-?\d+)? \+(\d+)", hunk_line_str).groups()[1])
-            lineno = lineno + additional_lines
-
-            self.load_file_at_line(filename, lineno)
-
-    def load_file_at_line(self, filename, lineno):
+        def first_per_file(items):
+            # type: (Iterator[Tuple[str, int, int]]) -> Iterator[Tuple[str, int, int]]
+            seen = set()  # type: Set[str]
+            for item in items:
+                filename, _, _ = item
+                if filename not in seen:
+                    seen.add(filename)
+                    yield item
+
+        diff = parse_diff_in_view(self.view)
+        jump_positions = filter_(self.jump_position_to_file(diff, pt) for pt in cursor_pts)
+        for jp in first_per_file(jump_positions):
+            self.load_file_at_line(*jp)
+
+    def load_file_at_line(self, filename, row, col):
+        # type: (str, int, int) -> None
         """
         Show file at target commit if `git_savvy.diff_view.target_commit` is non-empty.
         Otherwise, open the file directly.
         """
         target_commit = self.view.settings().get("git_savvy.diff_view.target_commit")
         full_path = os.path.join(self.repo_path, filename)
+        window = self.view.window()
+        if not window:
+            return
+
         if target_commit:
-            self.view.window().run_command("gs_show_file_at_commit", {
+            window.run_command("gs_show_file_at_commit", {
                 "commit_hash": target_commit,
                 "filepath": full_path,
-                "lineno": lineno
+                "lineno": row,
             })
         else:
-            self.view.window().open_file(
-                "{file}:{row}:{col}".format(file=full_path, row=lineno, col=0),
+            window.open_file(
+                "{file}:{row}:{col}".format(file=full_path, row=row, col=col),
                 sublime.ENCODED_POSITION
             )
 
+    def jump_position_to_file(self, diff, pt):
+        # type: (ParsedDiff, int) -> Optional[Tuple[str, int, int]]
+        head_and_hunk_offsets = head_and_hunk_for_pt(diff, pt)
+        if not head_and_hunk_offsets:
+            return None
+
+        header_region, hunk_region = head_and_hunk_offsets
+        header = self.view.substr(sublime.Region(*header_region))
+        hunk = self.view.substr(sublime.Region(*hunk_region))
+        hunk_start, _ = hunk_region
+
+        rowcol = real_rowcol_in_diff(hunk, *rowcol_in_hunk(self.view, hunk_start, pt))
+        if not rowcol:
+            return None
+
+        row, col = rowcol
+
+        filename = extract_filename_from_header(header)
+        if not filename:
+            return None
+
+        return filename, row, max(col, 1)
```

Comment:
another suggestion, how about we return a named tuple instead of regular tuple here? This would make it easier to refer to each jump position item's arguments without indexing / unpacking, and can still be passed as "star args" if want?